docs/concepts/fundamentals/render_api.md

```diff
@@ -49,42 +46,159 @@ rendered = Table.render(
+        assert self.args[0] == 123
+        assert self.args[1] == 10
+```
+
+## Kwargs
+
+The `kwargs` argument as passed to
+[`Component.get_template_data()`](../../../reference/api/#django_components.Component.get_template_data).
+
+If you defined the [`Component.Kwargs`](../../../reference/api/#django_components.Component.Kwargs) class,
+then the [`Component.kwargs`](../../../reference/api/#django_components.Component.kwargs) property will return an instance of that class.
+
+Otherwise, `kwargs` will be a plain dictionary.
+
+Raises `RuntimeError` if accessed outside of rendering execution.
+
+**Example:**
+
+With `Kwargs` class:
+
+```python
+from django_components import Component
+
+class Table(Component):
+    class Kwargs(NamedTuple):
+        page: int
+        per_page: int
+
+    def on_render_before(self, context: Context, template: Template) -> None:
+        assert self.kwargs.page == 123
+        assert self.kwargs.per_page == 10
+
+rendered = Table.render(
+    kwargs={"page": 123, "per_page": 10},
+)
+```
+
+Without `Kwargs` class:
+
+```python
+from django_components import Component
+
+class Table(Component):
+    def on_render_before(self, context: Context, template: Template) -> None:
+        assert self.kwargs["page"] == 123
+        assert self.kwargs["per_page"] == 10
+```
+
+## Slots
+
+The `slots` argument as passed to
+[`Component.get_template_data()`](../../../reference/api/#django_components.Component.get_template_data).
+
+If you defined the [`Component.Slots`](../../../reference/api/#django_components.Component.Slots) class,
+then the [`Component.slots`](../../../reference/api/#django_components.Component.slots) property will return an instance of that class.
+
+Otherwise, `slots` will be a plain dictionary.
+
+Raises `RuntimeError` if accessed outside of rendering execution.
+
+**Example:**
+
+With `Slots` class:
+
+```python
+from django_components import Component, Slot, SlotInput
+
+class Table(Component):
+    class Slots(NamedTuple):
+        header: SlotInput
+        footer: SlotInput
+
+    def on_render_before(self, context: Context, template: Template) -> None:
+        assert isinstance(self.slots.header, Slot)
+        assert isinstance(self.slots.footer, Slot)
+
+rendered = Table.render(
+    slots={
+        "header": "MY_HEADER",
+        "footer": lambda ctx: "FOOTER: " + ctx.data["user_id"],
+    },
+)
+```
+
+Without `Slots` class:
+
+```python
+from django_components import Component, Slot, SlotInput
+
+class Table(Component):
+    def on_render_before(self, context: Context, template: Template) -> None:
+        assert isinstance(self.slots["header"], Slot)
+        assert isinstance(self.slots["footer"], Slot)
 ```
 
 ## Component inputs
 
-All the component inputs are captured and available as [`self.input`](../../../reference/api/#django_components.Component.input).
+You can access the most important inputs via [`self.args`](../render_api/#args),
```

Comment:
Added the `Component.args`, `Component.kwargs`, and `Component.slots` to the docs. And also started referring to the old `Component.input` as "low-level" access, because it contains the args/kwargs/slots as plain lists/dicts.